### PR TITLE
Implement server side pagination

### LINKED ISSUE #23: Have a proper pagination
Currently we're using `react-lazyload` to workaround the need for pagination, this created a much bigger problem where we can't keep the scroll position after going back from read screen.

We should have a proper pagination.

### PR DESCRIPTION
**Pull request by huytd**
Test build: https://webuild-blog-aoud2ry43-kingbazoka.vercel.app/

What changed in this PR:

1. Implement a mem cache and server side pagination
2. Remove lazyload on the client side
3. Fixed the scroll position issue
4. Removed the top progress bar on navigation

Fixed #23

### CODE REVIEW

File: pages/read.tsx [hunk 1]
```diff
@@ -24,12 +23,21 @@ export const getServerSideProps = async (context) => {
 const ReadPage = ({ article }) => {
   const router = useRouter();
   const { url } = router.query;
+
+  const backButtonClickHandler = () => {
+    if (window.history.length > 1) {
+      router.back();
```
Comment (ZeroX-DG) [inline pages/read.tsx:29]: But why `router.back()` you go back to "trang chủ" anyway. Why waste time comparing `window.history.length`? :thinking:

**Comment by ZeroX-DG** [comment]
bugggggggggg
Position shifted! :cry:

File: pages/read.tsx (hunk 1 above)
Comment (TheAlemazing) [reply pages/read.tsx:29]: from slack conservation:
```
handle vậy là đúng cách đó, hàm router.back() nó gọi window.history.back(), như thế sẽ keep đc cái vị trí scroll của trang trước, còn cách ban đầu là jump tới địa chỉ "/", cách này trigger navigation bằng cách push 1 item mới vô history, và nó ko phải là jump back nên sẽ ko có chuyện keep đc scroll position```

Comment (huytd) [reply pages/read.tsx:29]: Good question. First, the reason we need to use `router.back()` is to go back in history (it called `window.history.back()` under the hood). Going this way the browser will handle the scroll position for us. In the previous way, navigate to home page at "/" is just a new push to the browser's history queue, so no scroll position will be perserved.

As for the history length check: We have 2 ways of navigate to a post content page, the first one is when the user open it directly in the home page (by clicking on the post's title, or click on the "Read more" button). The second one is when the user open the post content page in a new tab.

In the first case, go back in history is the obvious way. But in the second case, there will be no history queue, as it's the new window. So we just go to the home page for that case.

**Review by lqt93 — APPROVED** [review]
LGTM. The pagination can be fixed in #24 .

**Pull request merged**